Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1465,6 +1492,33 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  public override func visit(_ node: PoundSourceLocationSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+
+    if let args = node.args, args.fileName.hasError {
+      var rawDelimiters: [TokenSyntax] = []
+      if let unexpectedBeforeOpenQuote = args.fileName.unexpectedBeforeOpenQuote?.onlyPresentToken(where: { $0.tokenKind.isRawStringDelimiter }) {
+        rawDelimiters += [unexpectedBeforeOpenQuote]
+      }
+
+      if let unexpectedAfterCloseQuote = args.fileName.unexpectedAfterCloseQuote?.onlyPresentToken(where: { $0.tokenKind.isRawStringDelimiter }) {
+        rawDelimiters += [unexpectedAfterCloseQuote]
+      }
+
+      if !rawDelimiters.isEmpty {
+        addDiagnostic(
+          node,
+          ForbiddenExtendedEscapingString(token: args.fileLabel),
+          handledNodes: rawDelimiters.map { $0.id }
+        )
+      }
+    }
+
+    return .visitChildren
+  }
+
```

Comment:
Can’t we unify this with `visit(_ node: AvailabilityLabeledArgumentSyntax)` by diagnosing it in `visit(_ node: SimpleStringLiteralSyntax)`?

Reply:
The reason was that we using `args.fileLabel` to create the message. 
We can unify it by making a more generic error message maybe?

Reply:
Since this isn’t a super common error, I would prefer the generalized implementation at the expense of making the error message slightly more generic, e.g. just `argument cannot be an extended escaping string literal`